test/runner.js

```diff
@@ -12,46 +12,60 @@ const debug = require( 'debug' )( 'test-runner' ),
 	Mocha = require( 'mocha' ),
 	path = require( 'path' ),
 	program = require( 'commander' ),
+	watchFile = require( 'fs' ).watchFile,
 	chalk = require( 'chalk' );
 
-/**
- * Internal dependencies
- */
-const boot = require( './boot-test' ),
-	setup = require( './setup' );
-
 program
 	.usage( '[options] [files]' )
 	.option( '-R, --reporter <name>', 'specify the reporter to use', 'spec' )
 	.option( '-t, --node-total <n>', 'specify the node total to use', parseInt )
 	.option( '-i, --node-index <n>', 'specify the node index to use', parseInt )
-	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' );
+	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' )
+	.option( '-w, --watch', 'watch files for changes' );
 
 program.name = 'runner';
 
 program.parse( process.argv );
 
-const mocha = new Mocha( {
-	ui: 'bdd',
-	reporter: program.reporter
-} );
+const getMocha = function() {
+	/* These internal dependencies are not at the top of the file
+	 * because we need to re-require them every time we invalidate the cache.
+	 *
+	 * We invalidate the cache because Mocha has trouble running a test-suite
+	 * more than once:  https://github.com/mochajs/mocha/issues/995
+	 */
+	Object.keys( require.cache ).forEach( ( key ) => {
```

Comment:
You  mean that purging cache is not necessary on the first run?  While I agree of course, purging the cache each time makes the code more simple.  I ran it both ways with the time command to see if there was a noticeable speed difference.  The random difference in successive runs was more significant than the difference caused by cache-purging.   In order to get a more accurate picture for if there is a statistically significant performance hit I would need to grab the avg over many runs (seems like overkill / over-optimization). 

Given the research I performed, I would prefer to run the cache-purge every run for simplicity.
 

Here were some sample times on my 2013 mac-air:

Without purging cache:

<pre> 
real	0m5.166s
user	0m4.712s
sys	    0m0.554s
</pre>

With purging cache:
<pre> 
real	0m5.187s
user	0m4.755s
sys	    0m0.520s
</pre>